#167749956 Implement user profile settings

ldonjibson opened the pull request:
**What does this PR do?**
This PR implement user profile settings

**Description of Tasks to be completed**
- Add more field to user model and migration
- Write validations for newly added fields using joi
- Create controller function update newly added field 
- Utilize generated token from signin/signup as part of the logic 
   to make sure only the profile owner is able to update profile
- Write two routes a patch(to update) and a get request(to retrieve)
- Update Swagger documentation to update and retrieve user data

**How should this be manually tested?**
- Clone the Repo, change into the directory and run npm install
- Run npm run dev to start the server
- Make a PATCH/GET request to http://localhost:{PORT}/api/v1/user/profile/{email} on Postman using valid token as header **or** swagger at http://localhost:{PORT}/docs

Sample parameter
`24hours@gmail.com`

Sample request body
```
{
        "firstName": "Vio9la",
        "lastName": "Vi8olin",
        "email": "mathyr@gmail.com",
        "gender": "Male",
        "birthDate": "1980-06-25",
        "preferredLanguage": "English",
        "address": "22,juwon adedokun street",
        "department": "Administration",
        "lineManager": "Pius",
        "phoneNumber": "08181384092",
}
```
Relevant pivotal tracker story
[#167749956](https://www.pivotaltracker.com/story/show/167749956) 

Screenshot
![Screenshot002](https://user-images.githubusercontent.com/14142885/64224308-017c8280-cecf-11e9-8ef6-ac188dc5e8e3.png)

![Screenshot003](https://user-images.githubusercontent.com/14142885/64224159-65eb1200-cece-11e9-9da8-8a04d900f6ed.png)

alatos2 commented on src/database/migrations/20190824191528-create-users.js:
```diff
@@ -0,0 +1,29 @@
+    },
+    birthDate: { type: Sequelize.DATEONLY, allowNull: true },
+    prefferedLanguage: { type: Sequelize.STRING, allowNull: true },
+    prefferedCurrency: { type: Sequelize.STRING, allowNull: true },
```
You have done a nice job. Just to remind you that prefferedCurrency won't be needed again as agreed last week Friday. Keep up the good work. Also, kindly correct **prefferedLanguage** to **preferredLanguage**

ldonjibson replied:
@alatos2 thanks for the observation.

ldonjibson replied:
changes made

ldonjibson committed:
feat(user-profile): user can view/update profile

- add new fields to the existing ones to get more information from the
  user after signup
- create validations for each field so that each field can be independent
  of each other
- add more fields tothe user model- add more fields tothe user model
- create route for user to be able to update profile using patch request
  method
- create route for user to be able to view profile before or after update
  using the get method.
- write test for each both endpoints to make sure most of the real cases
  is covered

[Delivers #167749956]
